src/starterbot/BaseManager.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "Tools.h"
+#include "UnitManager.h"
+#include <unordered_map>
+
+// A Base struct is created for every resource cluster on the map. Each base holds references
+// to the resources and buildings placed within it, and information on whether the enemy is
+// currently occupying it.
+struct Base {
+
+	// The weighted center of the resources in this base.
+	bw::Position centroid;
+
+	// The minerals and geysers that belong to this base.
+	std::vector<bw::Unit> minerals;
+	std::vector<bw::Unit> geysers;
+
+	// All of the buildings closest to this base.
+	std::vector<bw::Unit> buildings;
```

Comment:
These should all be `bw::Unitset` so they can be filtered with e.g. `UnitManager::matchUnits()` and checked for membership easily without using costly `std::find` operations.